sway-parse/src/item/mod.rs

```diff
@@ -20,67 +19,66 @@ mod item_use;
 
 impl Parse for ItemKind {
     fn parse(parser: &mut Parser) -> ParseResult<ItemKind> {
-        if parser.peek::<UseToken>().is_some() || parser.peek2::<PubToken, UseToken>().is_some() {
+        // FIXME(Centril): Visibility should be moved out of `ItemKind` variants,
+        // introducing a struct `Item` that holds the visibility and the kind,
+        // and then validate in an "AST validation" step which kinds that should have `pub`s.
```

Comment:
In general, we try to keep a tracking issue for each TODO/FIXME and link it inline. Maybe we should add one and link it here? Edit: Woops, forgot to add this with previous review.